src/main/java/io/confluent/maven/resolver/ResolveKafkaVersionRangeMojo.java

```diff
@@ -0,0 +1,239 @@
+package io.confluent.maven.resolver;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+import org.codehaus.plexus.util.StringUtils;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.repository.RemoteRepository;
+import org.eclipse.aether.resolution.VersionRangeRequest;
+import org.eclipse.aether.resolution.VersionRangeResolutionException;
+import org.eclipse.aether.resolution.VersionRangeResult;
+import org.eclipse.aether.version.Version;
+
+/**
+ * This is a customized solution to resolves ce/ccs kafka version range to the
+ * highest matching version. By default, snapshots are excluded, as a workaround
+ * for <a href="https://issues.apache.org/jira/browse/MNG-3092">MNG-3092</a>.
+ *
+ * <p>
+ * For the version range format, see <a href=
+ * "https://github.com/eclipse/aether-core/blob/master/aether-util/src/main/java/org/eclipse/aether/util/version/GenericVersionScheme.java">
+ * GenericVersionScheme</a>
+ * </p>
+ *
+ * <p>
+ * Command-line example:
+ * </p>
+ *
+ * <pre>
+ * mvn io.confluent:resolver-maven-plugin:resolve-kafka-range \
+ *     -Dresolve.groupId=org.apache.kafka -Dresolve.artifactId=kafka-client "-Dresolve.versionRange=[6.0.0-1, 6.0.1-1]" \
```

Comment:
I figure that we actually do not need to add ce ccs qualifier? I mean the plugin will filter any version without ce or ccs. So kafka can share the same version range with other CP components.

Reply:
i don't follow how using a range with out the ce or ccs qualifier would even return you a result that did include artifacts with those qualifiers. either way I think we should still include the qualifiers in the range properties because it would be confusing to people to look in a pom file and see a range like that when they know the artifacts all contain the ce or ccs qualifier. it would end being some mystery to people as to why it worked.